apps/backend/backend_plugin.c

```diff
@@ -848,6 +862,8 @@ plugin_transaction_commit_all(clixon_handle       h,
     while ((cp = clixon_plugin_each(h, cp)) != NULL) {
         i++;
         if (plugin_transaction_commit_one(cp, h, td) < 0){
+            /*First make an effort ro revert transaction for the failed plugin*/
+            plugin_transaction_revert_failed(cp, h, td);
```

Comment:
Since this is called after a commit failed, maybe the name of the callback should be "commit-failed"?
Also, maybe one should handle the return value of the call, could it be that one wants to stop the following plugin_transaction_revert_all() call, or do you want it to continue uninterrupted?
(Minor: please have an initial space in the comment: `/*First make an` --> /`* First make an`

Reply:
@olofhagsand ,
`Since this is called after a commit failed, maybe the name of the callback should be "commit-failed"?`
Agree w/ you, will rename.
`Also, maybe one should handle the return value of the call, could it be that one wants to stop the following plugin_transaction_revert_all() call, or do you want it to continue uninterrupted?`
I believe it's better to follow the same logic as normal revert. The difference should be in the code placed inside this callback. Developer have to consider if the changes within transaction data has been succesfully applied to the underlying system during commit or not and incrementally apply reverse rules and operations. But all of this is up to developer and customization features.
`(Minor: please have an initial space in the comment: /*First make an --> /* First make an`
yep, will do

Reply:
Hi @olofhagsand,
I've made changes discussed above, pls check.